dotCMS/src/integration-test/java/com/dotcms/security/secret/ServiceIntegrationAPIImplTest.java

```diff
@@ -0,0 +1,344 @@
+        IntegrationTestInitService.getInstance().init();
+        SecretsStore.INSTANCE.get().deleteAll();
+    }
+
+    @Test
+    public void Test_Store_Json_Then_Recover_Decrypted_Then_Delete_Then_Verify_Its_Gone()
+            throws DotDataException, DotSecurityException {
+        final User admin = TestUserUtils.getAdminUser();
+        ServiceSecrets.Builder builder = new ServiceSecrets.Builder();
+        final String serviceKey = "anyServiceKey";
+        final ServiceSecrets bean = builder.withServiceKey(serviceKey)
+                .withHiddenSecret("mySecret1", "Once I saw a UFO")
+                .withHiddenSecret("mySecret2", "Football soccer sucks!")
+                .withSecret("boolSecret1", true)
+                .withSecret("boolSecret2", false)
+                .build();
+
+        final Host host = new SiteDataGen().nextPersisted();
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        api.saveSecrets(bean, host, admin);
+        final Optional<ServiceSecrets> optionalBean = api
+                .getSecrets(serviceKey, host, admin);
+
+        assertTrue(optionalBean.isPresent());
+
+        final ServiceSecrets recoveredBean = optionalBean.get();
+        assertEquals(serviceKey, recoveredBean.getServiceKey());
+        assertTrue(recoveredBean.getSecrets().containsKey("mySecret1"));
+        assertTrue(recoveredBean.getSecrets().containsKey("mySecret2"));
+        assertTrue(recoveredBean.getSecrets().containsKey("boolSecret1"));
+        assertTrue(recoveredBean.getSecrets().containsKey("boolSecret2"));
+
+        final Secret secret1 = recoveredBean.getSecrets().get("mySecret1");
+        assertEquals("Once I saw a UFO", secret1.getString());
+        assertTrue(secret1.isHidden());
+
+        final Secret secret2 = recoveredBean.getSecrets().get("mySecret2");
+        assertEquals("Football soccer sucks!", secret2.getString());
+        assertTrue(secret2.isHidden());
+
+        final Secret boolSecret1 = recoveredBean.getSecrets().get("boolSecret1");
+        assertTrue(boolSecret1.getBoolean());
+        assertFalse(boolSecret1.isHidden());
+
+        final Secret boolSecret2 = recoveredBean.getSecrets().get("boolSecret2");
+        assertFalse(boolSecret2.getBoolean());
+        assertFalse(boolSecret2.isHidden());
+    }
+
+    @Test
+    public void Test_Store_Json_On_Different_Sites_Then_Pull_Sites_By_Host_Verify_Match()
+            throws DotDataException, DotSecurityException {
+
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host1 = new SiteDataGen().nextPersisted();
+        final Host host2 = new SiteDataGen().nextPersisted();
+
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+
+        final ServiceSecrets bean1Host1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("h1:secret1", "sec1")
+                .withHiddenSecret("h1:secret2", "sec2")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "sec2" appears 4 times in this file; the first occurrence is on line 87](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)